pvanalytics/quality/irradiance.py

```diff
@@ -471,3 +471,177 @@ def daily_insolation_limits(irrad, clearsky, daily_min=0.4, daily_max=1.25):
+    component_sum_df = pd.DataFrame({'ghi': ghi,
+                                     'dhi': dhi,
+                                     'dni': dni})
+    return component_sum_df
+
+
+def calculate_component_sum_series(solar_zenith,
+                                   ghi=None,
+                                   dhi=None,
+                                   dni=None,
+                                   dni_clear=None,
+                                   zenith_limit=90,
+                                   fill_night_value=np.nan):
+    '''
+    Use the component sum equations to calculate the missing series, using
+    the other available time series. One of the three parameters (ghi, dhi,
+    dni) is passed as None, and the two series are used to
+    calculate the missing series. After calculation, the series is
+    run through a nighttime routine, where nighttime values are set based on
+    the fill_night_value parameter.
+
+    The "component sum" or "closure" equation relates the three
+    primary irradiance components as follows:
+    .. math::
+       GHI = DHI + DNI * cos(theta_z)
+
+    Parameters
+    ----------
+    solar_zenith : Series
+        Zenith angles in decimal degrees, with datetime index.
+        Angles must be >=0 and <=180. Must have the same datetime index
+        as ghi, dhi, and dni series, when available.
+    ghi : Series, optional
+        Pandas series of GHI data, with datetime index. Must have the same
+        datetime index as dni, dhi, and solar_zenith series, when available.
+    dhi : Series, optional
+        Pandas series of dni data, with datetime index. Must have the same
+        datetime index as ghi, dni, and zenith series, when available.
+    dni : Series, optional
+        Pandas series of dni data, with datetime index. Must have the same
+        datetime index as ghi, dhi, and zenith series, when available.
+    dni_clear : Series, optional
+        Pandas series of clearsky dni data. Must have the same datetime index
+        as ghi, dhi, dni, and zenith series, when available. See
+        :py:func:`dni` for details.
+    zenith_limit: Float
+        Solar zenith boundary between night and day, in degrees.
+        For calculation of the component sum, solar_zenith is set to 90 where
+        solar_zenith > zenith_limit.
+    fill_night_value: String or float or int, default np.nan
+        Options include 'equation', float or int values (np.nan, 0, etc.).
+        This is the fill value for nighttime periods.
+        If a float or int value is passed (np.nan, 0 , -.5, etc.), then
+        nighttime values are filled using the fill_night_value parameter.
+        If 'equation' is used, nighttime fill periods are based on the
+        component sum equation for night (where DNI is 0):
+            GHI = 0 + DHI
+        If neither of the above options are chosen, then the computed
+        nighttime values based of the component sum equation outpus are
+        returned.
```

Comment:
```suggestion
        If neither of the above options are chosen, then the computed
        nighttime values based of the component sum equation outpus are
        returned.
```
With `fill_night_value` default to `np.nan` this case appears to be excluded. The private function accepts anything which seems risky. What about `fill_night_value=None` being the default and the private function responds by filling with the component sum equation values (DNI being whatever it is)? If you agree, then add None as a permitted value.

I also think the private function should raise an Error if `fill_night_value` is not 'equation', float, int, np.nan or None.

Reply:
```
    if isinstance(fill_night_value, float) | isinstance(fill_night_value, int):
        # Replace the nighttime values with a fill value--this can be np.nan,
        # which is a float
        series[mask] = fill_night_value
    elif fill_night_value == 'equation':
        # Use the nighttime equation GHI = 0 + DHI, which translates as:
        # GHI_Calc (at night) = DHI_measured
        # DHI_Calc (at night) = GHI_measured
        # DNI_Calc (at night) = 0
        if component == 'GHI':
            series[mask] = component_sum_df['dhi'][mask]
        elif component == 'DHI':
            series[mask] = component_sum_df['ghi'][mask]
        else:
            series[mask] = 0
    elif fill_night_value is None:
        pass
    else:
        raise ValueError("The fill_night_value variable must be None,"
                         " float or int, or 'equation'. Please change "
                         "the variable fill_night_value value.")
```

I reworked the logic to throw an error if np.nan (float), float/int, 'equation' or None is not passed. I changed the default setting to None, where we don't do anything, just pass the component-sum calculated series back.